src/browser/api/system.js

```diff
@@ -392,36 +359,41 @@ exports.System = {
     getMousePosition: function() {
         return MonitorInfo.getMousePosition();
     },
-    getProcessList: function() {
-
-        let allApps = coreState.getAllApplications();
-        let runningAps = allApps.filter(app => {
-            return app.isRunning;
-        });
-
-        let processList = runningAps.map(app => {
-            var appObj = coreState.getAppObjByUuid(app.uuid),
-                name = appObj._options.name,
-                proc = appObj._processInfo || defaultProc;
 
-            return {
-                cpuUsage: proc.getCpuUsage(),
-                name: name,
-                nonPagedPoolUsage: proc.getNonPagedPoolUsage(),
-                pageFaultCount: proc.getPageFaultCount(),
-                pagedPoolUsage: proc.getPagedPoolUsage(),
-                pagefileUsage: proc.getPagefileUsage(),
-                peakNonPagedPoolUsage: proc.getPeakNonPagedPoolUsage(),
-                peakPagedPoolUsage: proc.getPeakPagedPoolUsage(),
-                peakPagefileUsage: proc.getPeakPagefileUsage(),
-                peakWorkingSetSize: proc.getPeakWorkingSetSize(),
-                processId: appObj.mainWindow.webContents.processId,
-                uuid: appObj.uuid,
-                workingSetSize: proc.getWorkingSetSize()
-            };
-        });
-
-        return processList;
+    getProcessList: function() {
+        let localConfig = this.getConfig();
+        let { 'runtime': { version: runtimeVersion }, 'startup_app': { uuid, name } } = localConfig && localConfig.data;
+
+        return electronApp
+            .getAppMetrics()
+            .filter(proc => proc.type.toLowerCase() !== 'utility') // results with 'utility' type are not actual processess
+            .map(proc => {
+                let procInfo = new electron.processInfo(proc.pid);
+
+                return procInfo ? {
+                    uuid,
```

Comment:
Won't this be the same (uuid/name) for every item in the response then? It should be per app that is responsible for the process